> The following passes tests, too.

Have you tried `./bin/doctest integrals` ?
(The Failed case is in Example docs). It Fails on my system.

```
__________________ sympy.integrals.transforms._simplifyconds ___________________
File "sympy/integrals/transforms.py", line 881, in sympy.integrals.transforms._simplifyconds
Failed example:
    simp(abs(x**2) < 1, x, 0)
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python2.7/doctest.py", line 1315, in __run
        compileflags, 1) in test.globs
      File "<doctest sympy.integrals.transforms._simplifyconds[5]>", line 1, in <module>
        simp(abs(x**2) < 1, x, 0)
      File "sympy/integrals/transforms.py", line 948, in _simplifyconds
        expr = repl(expr, StrictLessThan, replie)
      File "sympy/integrals/transforms.py", line 947, in repl
        return ex.replace(*args)
      File "sympy/core/basic.py", line 1351, in replace
        rv = bottom_up(self, rec_replace, atoms=True)
      File "sympy/simplify/simplify.py", line 4083, in bottom_up
        rv = F(rv)
      File "sympy/core/basic.py", line 1336, in rec_replace
        new = _value(expr, result)
      File "sympy/core/basic.py", line 1280, in <lambda>
        _value = lambda expr, result: value(*expr.args)
      File "sympy/integrals/transforms.py", line 933, in replie
        r = bigger(x, y)
      File "sympy/integrals/transforms.py", line 919, in bigger
        return bigger(1/ex2, 1/ex1)
      File "sympy/integrals/transforms.py", line 925, in bigger
        if n < 0 and (abs(ex1) >= abs(a)**n) == True:
      File "sympy/core/numbers.py", line 1806, in __ge__
        return Rational.__ge__(self, other)
      File "sympy/core/numbers.py", line 1457, in __ge__
        return Expr.__ge__(expr, other)
      File "sympy/core/expr.py", line 240, in __ge__
        raise TypeError("Invalid comparison of complex %s" % me)
    TypeError: Invalid comparison of complex zoo
```

Probably you have tried `sympy/integrals/tests/test_transforms.py` .

--- a/sympy/core/expr.py
+++ b/sympy/core/expr.py
@@ -235,7 +235,8 @@ class Expr(Basic, EvalfMixin):
         except SympifyError:
             raise TypeError("Invalid comparison %s >= %s" % (self, other))
         for me in (self, other):
-            if me.is_complex and me.is_real is False:
+            if (me.is_complex and me.is_real is False) or \
+                    me.has(S.ComplexInfinity):
                 raise TypeError("Invalid comparison of complex %s" % me)
             if me is S.NaN:
                 raise TypeError("Invalid NaN comparison")
@@ -252,7 +253,8 @@ class Expr(Basic, EvalfMixin):
         except SympifyError:
             raise TypeError("Invalid comparison %s <= %s" % (self, other))
         for me in (self, other):
-            if me.is_complex and me.is_real is False:
+            if (me.is_complex and me.is_real is False) or \
+                    me.has(S.ComplexInfinity):
                 raise TypeError("Invalid comparison of complex %s" % me)
             if me is S.NaN:
                 raise TypeError("Invalid NaN comparison")
@@ -269,7 +271,8 @@ class Expr(Basic, EvalfMixin):
         except SympifyError:
             raise TypeError("Invalid comparison %s > %s" % (self, other))
         for me in (self, other):
-            if me.is_complex and me.is_real is False:
+            if (me.is_complex and me.is_real is False) or \
+                    me.has(S.ComplexInfinity):
                 raise TypeError("Invalid comparison of complex %s" % me)
             if me is S.NaN:
                 raise TypeError("Invalid NaN comparison")
@@ -286,7 +289,8 @@ class Expr(Basic, EvalfMixin):
         except SympifyError:
             raise TypeError("Invalid comparison %s < %s" % (self, other))
         for me in (self, other):
-            if me.is_complex and me.is_real is False:
+            if (me.is_complex and me.is_real is False) or \
+                    me.has(S.ComplexInfinity):
                 raise TypeError("Invalid comparison of complex %s" % me)
             if me is S.NaN:
                 raise TypeError("Invalid NaN comparison")

--- a/sympy/core/tests/test_relational.py
+++ b/sympy/core/tests/test_relational.py
@@ -455,8 +455,15 @@ def test_nan_inequality_raise_errors():
 def test_nan_complex_inequalities():
     # Comparisons of NaN with non-real raise errors, we're not too
     # fussy whether its the NaN error or complex error.
-    for r in (I, Symbol('z', imaginary=True)):
+    for r in (I, zoo, Symbol('z', imaginary=True)):
         assert_all_ineq_raise_TypeError(r, nan)
+
+
+def test_complex_infinity_inequalities():
+    raises(TypeError, lambda: zoo > 0)
+    raises(TypeError, lambda: zoo >= 0)
+    raises(TypeError, lambda: zoo < 0)
+    raises(TypeError, lambda: zoo <= 0)
 
 
 def test_inequalities_symbol_name_same():

--- a/sympy/integrals/transforms.py
+++ b/sympy/integrals/transforms.py
@@ -916,7 +916,10 @@ def _simplifyconds(expr, s, a):
         if ex2.func is Abs:
             ex2 = ex2.args[0]
         if ex1.has(s):
-            return bigger(1/ex2, 1/ex1)
+            try:
+                return bigger(1/ex2, 1/ex1)
+            except TypeError:
+                return None
         n = power(ex2)
         if n is None:
             return None

